### DISCUSSION TITLE
String to texture conversion (blurry display)

### DISCUSSION THREAD

**Discussion by guillaC** [raysan5/raylib#3675] [Q&A]
Hello everyone,
I want each of my textures to contain a letter from a 2D array.
I am using RayLib via a .NET wrapper, so I have declared my arrays as follows:
```
    static char[,] tableau = new char[numRows, numColumns];
    static Texture2D[,] textures = new Texture2D[numRows, numColumns];
```
I fill in my Texture2D array like this:
```
   static void GenererTextures()
    {
        for (int i = 0; i < numRows; i++)
        {
            for (int j = 0; j < numColumns; j++)
            {
                Image image = ImageTextEx(fontGB, tableau[i, j].ToString(), textSize, 0.0f, Color.LIME);
                textures[i, j] = Raylib.LoadTextureFromImage(image);
                Raylib.UnloadImage(image);
            }
        }
    }
```
I display it like this:
```
    static void AfficherGrille()
    {
        for (int i = 0; i < numRows; i++)
        {
            for (int j = 0; j < numColumns; j++)
            {
                Texture2D texture = textures[i, j];
                Rectangle rect = new Rectangle(j * texture.Width, i * texture.Height, texture.Width, texture.Height);
                Raylib.DrawTexture(texture, (int)rect.X + 5, (int)rect.Y + 5, Color.YELLOW);
            }
        }
    }
```
And I obtain the result:

![image](https://github.com/raysan5/raylib/assets/6315083/1cb07301-b752-4f94-b4f4-ef693b14b983)
For your information, the font used is: https://www.dafont.com/fr/pixelated.font

How can I achieve a higher quality display? (my letters are a bit blurry..)

Best regards

**Comment by orcmid** [comment]
Is that display PNG Actual size?

What is the resolution of your display and have you set some sort of scaling factor?

**Reply by guillaC** [reply]
```
    const int screenWidth = 800;
    const int screenHeight = 600;
    const int cellSize = 10;
    const int textSize = 10;
```

I wanted to have a letter in size 10 within a square of size 10x10 as well.

**Reply by orcmid** [reply]
I meant characteristics of your device's display, including dpi :).

And which operating system are you building for?

**Reply by guillaC** [reply]
![image](https://github.com/raysan5/raylib/assets/6315083/26dad324-b889-4326-95fe-97bbe33e60f7)
![image](https://github.com/raysan5/raylib/assets/6315083/810c813c-a653-4740-8b16-b37403d24a07)
![image](https://github.com/raysan5/raylib/assets/6315083/b7570dc7-bac1-4794-8cea-dc43419dc48f)

**Comment by JupiterRider** [comment]
@guillaC Use `LoadFontEx` and try using different font sizes.
